### PR TITLE
[api] Don't use the redefined find_by_name, somehow this messes up qu…

### PR DESCRIPTION
**Pull request by hennevogel**
…eries after a while

### CODE REVIEW

File: src/api/app/models/project.rb [hunk 1]
```diff
@@ -341,7 +341,8 @@ def self.find_remote_project(name, skip_access=false)
         # hmm calling a private class method is not the best idea..
         lpro = nil # FIXME2.4
       else
-        lpro = Project.find_by_name(local_project, select: 'id,name,remoteurl')
```
Comment (bgeuken) [inline src/api/app/models/project.rb]: You skipped the select part. Was that on purpose?

Comment (hennevogel) [inline src/api/app/models/project.rb]: Yes it was

File: src/api/app/models/project.rb [hunk 2]
```diff
@@ -341,7 +341,8 @@ def self.find_remote_project(name, skip_access=false)
...
+        lpro = Project.find_by(name: local_project)
+        lpro = nil unless check_access?(lpro)
         logger.debug "Found local project #{local_project} with remoteurl #{lpro[:remoteurl]}" if lpro
```
Comment (bgeuken) [inline src/api/app/models/project.rb]: Mind to change this to
```ruby
if check_access?(lpro)
  logger.debug "Found local project #{local_project} with remoteurl #{lpro[:remoteurl]}"
else
  lpro = nil
end
```

**Comment by bgeuken** [comment]
LGTM

**Comment by hennevogel** [comment]
I've looked at it again. Please review

**Comment by bgeuken** [comment]
LGTM

**Pull request merged**